internal/platform/platform.go

```diff
@@ -25,11 +27,73 @@ func NewPlatformInfo() *PlatformInfo {
 	return &PlatformInfo{
 		Detectors: []VendorDetector{
 			NewIntelDetector(),
+			NewMarvellDetector(),
 			// add more detectors here
 		},
 	}
 }
 
+func (pi *PlatformInfo) Getvendorname() (string, error) {
+	klog.Infof("Detecting  Platform is DPU or not")
+	for _, detector := range pi.Detectors {
+		isDpu, err := detector.IsDpuPlatform()
+		if err != nil {
+			return "", err
```

Comment:
We need a way to detect the Vendor in the case we are running on the DPU as well.

Reply:
Yes, its supported.